core/src/main/java/brooklyn/entity/rebind/PeriodicDeltaChangeListener.java

```diff
@@ -494,5 +496,8 @@ public synchronized void onChanged(BrooklynObject instance) {
     public PersistenceExceptionHandler getExceptionHandler() {
         return exceptionHandler;
     }
-    
+
+    protected boolean shouldLogCheckpoint() {
+        return (checkpointLogCount.get() < 5) || (checkpointLogCount.get() % 1000 == 0);
```

Comment:
Did you intend to use incrementAndGet? Nothing increments the counters at the moment so they'll always be 0.

Also suggest getting the value of the integer once so we don't pay synchronisation penalty twice.